src/utils/DateUtils.ts

```diff
@@ -29,138 +157,230 @@ class DefaultDateToStringOptions extends Options implements IDateToStringOptions
+      return dateOnly.format('dddd, MMMM DD, YYYY');
     }
 
-    return Globalize.format(dateOnly, 'd');
+    return dateOnly.format('M/D/YYYY');
   }
 
+  /**
+   * Converts the time information from a date into a string.
+   * @param date The date containing the time information to format.
+   * @param options Specifies the enabled options on the date.
+   * @returns {any} Will either be an empty string or a string containing the time.
+   */
   static timeToString(date: Date, options?: IDateToStringOptions): string {
     if (Utils.isNullOrUndefined(date)) {
       return '';
     }
-
-    return Globalize.format(date, 't');
+    return moment(date).format('h:mm A');
   }
 
+  /**
+   * Converts the date into a string according to the specified options.
+   * This method uses [ `timeToString` ]{@link DateUtils.timeToString}, which adds a time information to the
+   * converted date. To display a date without a timestamp, use [ `dateToString` ]{@link DateUtils.timeToString}.
+   * @param date The date to convert to a string.
+   * @param options Specifies the enabled options on the date.
+   * @returns {any} The date after being formatted into a string according to the specified options.
+   */
   static dateTimeToString(date: Date, options?: IDateToStringOptions): string {
+    DateUtils.setLocale();
+
     if (Utils.isNullOrUndefined(date)) {
       return '';
     }
 
     options = new DefaultDateToStringOptions().merge(options);
+    const today = DateUtils.keepOnlyDatePart(options.now);
+    const isThisWeek = moment(date).diff(moment(today), 'weeks') == 0;
+    const datePart = DateUtils.dateToString(date, options);
+    const dateWithoutTime = DateUtils.keepOnlyDatePart(date);
 
-    var today = DateUtils.keepOnlyDatePart(options.now);
-    var isThisWeek = Math.abs(TimeSpan.fromDates(date, today).getDays()) < 7;
-    var datePart = DateUtils.dateToString(date, options);
-    var dateWithoutTime = DateUtils.keepOnlyDatePart(date);
-
-    if (options.alwaysIncludeTime || (options.includeTimeIfThisWeek && isThisWeek) || (options.includeTimeIfToday && dateWithoutTime.valueOf() == today.valueOf())) {
+    if (moment(date).isValid() && (options.alwaysIncludeTime == true || (options.includeTimeIfThisWeek == true && isThisWeek)
+      || (options.includeTimeIfToday == true && dateWithoutTime.valueOf() == today.valueOf()))) {
       return datePart + ', ' + DateUtils.timeToString(date);
     } else {
       return datePart;
     }
   }
 
+  /**
+   * Converts a month from a number to a string (e.g '0' to 'january').
+   * @param month The month represented by its number (0 to 11).
+   * @returns {string} The month's name associated to its number.
+   */
   static monthToString(month: number): string {
-    var date = new Date(1980, month);
-    return Globalize.format(date, 'MMMM');
+    DateUtils.setLocale();
+    const date = moment(new Date(1980, month)).toDate();
+    return moment(date).format('MMMM');
   }
 
```

Comment:
> Validates whether a value is an instance of Date.
> @param date The value to verify.
> @returns {boolean} `true` if the `date` argument is an instance of Date; `false` otherwise.